charger/easee.go

```diff
@@ -734,67 +798,13 @@ func (c *Easee) Phases1p3p(phases int) error {
 	return err
 }
 
-var _ api.Meter = (*Easee)(nil)
-
-// CurrentPower implements the api.Meter interface
-func (c *Easee) CurrentPower() (float64, error) {
-	if status, err := c.Status(); err != nil || status == api.StatusA {
-		return 0, err
-	}
-
-	c.mux.Lock()
-	defer c.mux.Unlock()
-
-	return c.currentPower, nil
-}
+var _ api.PhaseGetter = (*Easee)(nil)
 
-func (c *Easee) requestLifetimeEnergyUpdate() {
-	c.lastEnergyPollMux.Lock()
-	defer c.lastEnergyPollMux.Unlock()
-	if time.Since(c.lastEnergyPollTriggered) > time.Minute*3 { // api rate limit, max once in 3 minutes
-		uri := fmt.Sprintf("%s/chargers/%s/commands/%s", easee.API, c.charger, easee.PollLifetimeEnergy)
-		if _, err := c.Post(uri, request.JSONContent, request.MarshalJSON(nil)); err != nil {
-			c.log.WARN.Printf("Failed to trigger an update of LIFETIME_ENERGY: %v", err)
-		}
-		c.lastEnergyPollTriggered = time.Now()
-	}
-}
-
-var _ api.ChargeRater = (*Easee)(nil)
-
-// ChargedEnergy implements the api.ChargeRater interface
-func (c *Easee) ChargedEnergy() (float64, error) {
-	c.mux.Lock()
-	defer c.mux.Unlock()
-
-	// return either the self calced session energy (current LIFETIME_ENERGY minus remembered start value),
-	// or the SESSION_ENERGY value by the API. Each value could be lower than the other, depending on
-	// order and receive timestamp of the product update. We want to return the higher (and newer) value.
-	if c.sessionStartEnergy != nil {
-		return max(c.sessionEnergy, c.totalEnergy-*c.sessionStartEnergy), nil
-	}
-
-	return c.sessionEnergy, nil
-}
-
-var _ api.PhaseCurrents = (*Easee)(nil)
-
-// Currents implements the api.PhaseCurrents interface
-func (c *Easee) Currents() (float64, float64, float64, error) {
-	c.mux.Lock()
-	defer c.mux.Unlock()
-
-	return c.currentL1, c.currentL2, c.currentL3, nil
-}
-
-var _ api.MeterEnergy = (*Easee)(nil)
-
-// TotalEnergy implements the api.MeterEnergy interface
-func (c *Easee) TotalEnergy() (float64, error) {
-	c.mux.Lock()
-	defer c.mux.Unlock()
-
-	return c.totalEnergy, nil
+// GetPhases implements the api.PhaseGetter interface
+func (c *Easee) GetPhases() (int, error) {
+	c.mux.RLock()
+	defer c.mux.RUnlock()
+	return c.phaseMode, nil
```

Comment:
Ich kann es gerade nicht prüfen (frühestens Mittwoch) - aber ich glaube dieser Wert ist statisch und entspricht dem physikalischen Anschluß der Wallbox. Um den "aktuellen" Wert zu erhalten, müßten wir nach meinem Verständnis ObservationID 110 `OUTPUT_PHASES` lesen.

Reply:
@GrimmiMeloni I think if we change this we should also refactor the Easee a bit. Currently, there are lots of local variables plus `obsTime`. Imho we should add an `obs` `sync.Map` or mutex-guarded map that holds `Observation`s plus a `getObs(val any) error` method for accessing it. Wdyt?

Reply:
I double checked some recent logs of mine. They confirmed my hunch.
`PHASE_MODE` stays static (at 2 in my case), whereas `OUTPUT_PHASE` follows the active phases as controlled by evcc. Noteworthy implementation detail - `OUTPUT PHASE` values are not `0,1,3` but `0,10,30`, so we need to map (or simply div by 10).

I also like the suggested refactoring, but I would rather put this in a separate PR. It will touch a lot of code and would not want to mix this with this feature.

Reply:
Ok- then lets add another var here and the reduce all of them separately.

Reply:
Do you want me to take this over and do the rest (e.g. local testing, etc.)?